src/client/dfs/dfs.c

```diff
@@ -3061,7 +3065,7 @@ dfs_open_stat(dfs_t *dfs, dfs_obj_t *parent, const char *name, mode_t mode,
 	case S_IFLNK:
 		rc = open_symlink(dfs, parent, flags, cid, value, &entry, len,
 				  obj);
-		if (rc) {
+		if ((rc != 0) && (rc != EEXIST)) {
```

Comment:
i think it's fine to log a debug message of eexist

Reply:
A warning msg is generated in fs_copy_symlink() to handle EEXIST.

Reply:
yea as i said a D_DEBUG message is OK. a D_ERROR or D_WARN can be problemetic / confusing and i would agree to change that.
the problem is that you are making logic changes in the behavior of the DFS API (not freeing the value buffer; and returning success in open with O_CREAT in case of EEXIST).
what i think you want to do is only change:
https://github.com/daos-stack/daos/blob/master/src/client/dfs/dfs.c#L1080
to change D_ERROR to D_DEBUG